lib/simplecov/defaults.rb

```diff
@@ -4,7 +4,7 @@
 SimpleCov.profiles.define "root_filter" do
   # Exclude all files outside of simplecov root
   add_filter do |src|
-    !(src.filename =~ /^#{Regexp.escape(SimpleCov.root)}/i)
+    !src.filename.downcase.start_with?(SimpleCov.root.downcase)
```

Comment:
I am a bit surprised, regex solution is faster, but not because of the extraction and the `o` modifier but because of the `\A` change (which of course makes sense), but I did not expect that it will be faster than `String#start_with?`.

```
Calculating -------------------------------------
         start_with?   105.000  i/100ms
     match_optimized    14.000  i/100ms
   match_optimized_A   102.435k i/100ms
               match    13.000  i/100ms
-------------------------------------------------
         start_with?      1.110k (± 3.5%) i/s -      5.565k
     match_optimized    147.382  (± 6.1%) i/s -    742.000
   match_optimized_A      2.477M (± 6.6%) i/s -     12.395M
               match    143.284  (± 6.3%) i/s -    715.000

Comparison:
   match_optimized_A:  2477292.0 i/s
         start_with?:     1110.0 i/s - 2231.75x slower
     match_optimized:      147.4 i/s - 16808.64x slower
               match:      143.3 i/s - 17289.33x slower
```